NugetLightClient.cs

```diff
@@ -959,7 +959,7 @@ private static List<DependencyVersion> ReduceConstraints(List<DependencyVersion>
                     }
 
                     // get all the packages that match this dependency
-                    var dependentPackageItem = request.GetPackageById(dep.Id, request, minimumVersion: minVersion, maximumVersion: maxVersion, minInclusive: dep.DependencyVersion.IsMinInclusive, maxInclusive: dep.DependencyVersion.IsMaxInclusive).ToArray();
+                    var dependentPackageItem = request.GetPackageById(dep.Id, request, minimumVersion: minVersion, maximumVersion: maxVersion, minInclusive: dep.DependencyVersion.IsMinInclusive, maxInclusive: dep.DependencyVersion.IsMaxInclusive, isDependency: true).ToArray();
```

Comment:
By setting it true, NuGet provider will not return unlisted packages. Do we want to make this change for PowerShellGet only or generic, any repositories including NuGet.org?

Reply:
As I understand it, this mirrors the behavior of nuget so I think it should be generic

Reply:
The nuget provider will still return unlisted packages.  The “isDependency” param is used to determine whether an unlisted package should be added to the set of potential packages to be returned. 
The changes are meant to be generic, I’ve been testing with nuget and powershellgallery.

Reply:
The nuget provider will still return unlisted packages.  The “isDependency” param is used to determine whether an unlisted package should be added to the set of potential packages to be returned. 
The changes are meant to be generic, I’ve been testing with nuget and powershellgallery.